code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/InternalMessagingUtils.java

```diff
@@ -300,6 +301,28 @@ static boolean isSchemaConsequence(final Event event) {
         return consequenceType.equals(MessagingConstants.ConsequenceDetailKeys.SCHEMA);
     }
 
+    /**
+     * Determines if the passed in {@code Event} is an event history disqualify event.
+     *
+     * @param event An event history write {@link Event}.
+     * @return {@code boolean} indicating if the passed in event is an event history disqualify
+     *     event.
+     */
+    static boolean isEventHistoryDisqualifyEvent(final Event event) {
+        final Map<String, Object> eventData = event.getEventData();
+        final Map<String, Object> eventHistoryMap =
+                DataReader.optTypedMap(
+                        Object.class,
+                        eventData,
+                        MessagingConstants.EventDataKeys.IAM_HISTORY,
+                        null);
+        return MessagingEdgeEventType.DISQUALIFY
+                .getPropositionEventType()
+                .equalsIgnoreCase(
+                        DataReader.optString(
+                                eventHistoryMap, MessagingConstants.EventMask.Keys.EVENT_TYPE, ""));
```

Comment:
nit: Since `eventHistoryMap` can be null here, it might throw `DataReaderException`. Can we wrap it in a try-catch block?

Reply:
Actually nevermind the exception is caught by Core